extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala

```diff
@@ -63,45 +63,62 @@ class TransactionMultiTableSpec
+    transactions.map(_.ledger_offset).toSet should have size 3
   }
 
   "Exercises" should "be extracted" in {
-    getExercises should have length 1
+    getExercises should have length 2
   }
 
   "All the data" should "represent what went down in the scenario" in {
     // `transaction1` created `contract1`, then
     // `transaction2` created `exercise`, which archived `contract1` and resulted `contract2`
 
-    val List(transaction1, transaction2) = getTransactions.sortBy(_.seq)
-    val List(exercise) = getExercises
-    val List((archived_by_event_id1, transaction_id1, archived_by_transaction_id1)) =
+    val List(transaction1, transaction2, transaction3) = getTransactions.sortBy(_.seq)
+    val List(exercise1, exercise2) = getExercises
+    val List(
+      (archived_by_event_id1, transaction_id1, archived_by_transaction_id1),
+      (archived_by_event_id2, transaction_id2, archived_by_transaction_id2)) =
       getResultList[(Option[String], String, Option[String])](
         sql"SELECT _archived_by_event_id, _transaction_id, _archived_by_transaction_id FROM template.transactionexample_rightofuseoffer")
-    val List((event_id2, archived_by_event_id2, transaction_id2, archived_by_transaction_id2)) =
+    val List(
+      (
+        event_id_accept,
+        archived_by_event_id_accept,
+        transaction_id_accept,
+        archived_by_transaction_id_accept)) =
       getResultList[(String, Option[String], String, Option[String])](
         sql"SELECT _event_id, _archived_by_event_id, _transaction_id, _archived_by_transaction_id FROM template.transactionexample_rightofuseagreement")
 
     // `transaction1` created `contract1`, then
     transaction_id1 shouldEqual transaction1.transaction_id
 
-    // `transaction2` created `exercise`
-    exercise.transaction_id shouldEqual transaction2.transaction_id
+    // `transaction2` created `exercise1`
+    exercise1.transaction_id shouldEqual transaction2.transaction_id
 
-    // `exercised` archived `contract1`
+    // `exercise1` archived `contract1`
     archived_by_transaction_id1 shouldEqual Some(transaction2.transaction_id)
-    archived_by_event_id1 shouldEqual Some(exercise.event_id)
+    archived_by_event_id1 shouldEqual Some(exercise1.event_id)
 
     // ... while it resulted in `contract2`
-    exercise.child_event_ids.asArray.toList.toVector.flatten should contain(event_id2.asJson)
-    transaction_id2 shouldEqual transaction2.transaction_id
+    exercise1.child_event_ids.asArray.toList.toVector.flatten should contain(event_id_accept.asJson)
+    transaction_id_accept shouldEqual transaction2.transaction_id
     // which is not archived
-    archived_by_transaction_id2 shouldEqual None
-    archived_by_event_id2 shouldEqual None
+    archived_by_transaction_id_accept shouldEqual None
+    archived_by_event_id_accept shouldEqual None
+
+    // `transaction3` created `contract3`; then
+    transaction_id2 shouldEqual transaction3.transaction_id
```

Comment:
How does the comment relate to the check?

Reply:
The naming here is horrible:
The multitable schema creates one table per schema. `transaction_id2` is transaction id of the second offer contract which is the third contract in total. I’ve added the `offer` suffix consistently and a few comments which hopefully help.

Reply:
Thank you for clarifying, makes sense.